keel-sql/src/test/kotlin/com/netflix/spinnaker/keel/sql/TestContainer.kt

```diff
@@ -5,16 +5,17 @@ import org.jooq.SQLDialect.MYSQL
 import org.testcontainers.containers.JdbcDatabaseContainer
 import org.testcontainers.containers.MySQLContainerProvider
 
-internal fun initTestDatabase() = initDatabase(
-  mySQLContainer.authenticatedJdbcUrl,
-  MYSQL
-)
+internal val testDatabase by lazy {
+  initDatabase(mySQLContainer.authenticatedJdbcUrl, MYSQL)
+}
 
-private val mySQLContainer = MySQLContainerProvider()
+internal val mySQLContainer = MySQLContainerProvider()
   .newInstance("5.7.22")
   .withDatabaseName("keel")
   .withUsername("keel_service")
   .withPassword("whatever")
+  .withReuse(true)
+  .withNetwork(null)
```

Comment:
the `withReuse(true)` does the same as `TC_REUSABLE=true` for programatically created containers (e.g. most of the integration tests in `keel-sql`). I'm still trying to figure out how I can get both Spring and non-Spring tests to share the _same_ container. But at least this way we only start up 2.

Reply:
FYI I would recommend using `new MySQLContainer("mysql:5.7.22")` instead of `MySQLContainerProvider().newInstance("5.7.22")`, as `MySQLContainerProvider` is just for the JDBC infrastructure.

Also, unless the network is explicitly set earlier, `.withNetwork(null)` is not needed (even in Kafka starting with Testcontainers 1.15.x)

Reply:
>  I'm still trying to figure out how I can get both Spring and non-Spring tests to share the same container

Some implementation detail:
The reusable mode is using hashing. If both instances have the same hash, the container is reused, meaning that setting the same database name / username / password and other settings would result in the same hash.
Double check that _all_ parameters are the same.

Reply:
Also, since `mySQLContainer` variable is private, I got a feeling that you may as well use the JDBC-style integration and pass static `jdbc:tc:mysql:5.7.22:///keel&TC_REUSABLE=true` to `initDatabase`

Reply:
Yeah, I did try doing that and still ended up with multiple containers running (checking via `docker ps -a` afterward). I figured something was throwing the hash off but I wasn't able to pin it down.

Reply:
> FYI I would recommend using `new MySQLContainer("mysql:5.7.22")` instead of `MySQLContainerProvider().newInstance("5.7.22")`, as `MySQLContainerProvider` is just for the JDBC infrastructure.

It doesn't seem like I can call this constructor from Kotlin due to the recursive generic type. 
> 
> Also, unless the network is explicitly set earlier, `.withNetwork(null)` is not needed (even in Kafka starting with Testcontainers 1.15.x)

That was just a desperate attempt to see if that was the issue preventing the hashes from matching.

Reply:
> It doesn't seem like I can call this constructor from Kotlin due to the recursive generic type.

Yes, this is a known limitation of Kotlin, but you can use `<Nothing>` + `apply {}`:
https://github.com/testcontainers/testcontainers-java/blob/97493b112f76711adc8f431f6f8e6f3245f05c47/examples/spring-boot-kotlin-redis/src/test/kotlin/com/example/redis/AbstractIntegrationTest.kt#L21

> Yeah, I did try doing that and still ended up with multiple containers running (checking via docker ps -a afterward). I figured something was throwing the hash off but I wasn't able to pin it down.

With MySQL? Wow. I would highly appreciate something runnable so that I can debug & help figuring out why the hash wasn't the same! (Although I understand that you got it working and most probably already switched so something else 😅  I will try to reproduce it myself, but any help would be appreciated!)